### DISCUSSION TITLE
Hardening issue when using OPC DA as data source

### DISCUSSION THREAD

**Discussion by duduyoyo** [SCADA-LTS/Scada-LTS#2891] [Q&A]
If I understand correctly, Scada-LTS uses j-interop as an interoperable library between Java and COM. Based on the [link](https://sourceforge.net/p/j-interop/discussion/600729/thread/731d99133f/), j-interop hasn't solved the issue after MS has hardened OS for security [reason](https://support.microsoft.com/en-us/topic/kb5004442-manage-changes-for-windows-dcom-server-security-feature-bypass-cve-2021-26414-f1400b52-c141-43d2-941e-37ed901c769c). So, is Scada-LTS capable of communicating with OPC DA properly with a hardened OS? Thanks!

**Comment by Limraj** [comment]
Hi @duduyoyo,
I confirm that we use the following library to support OPC DA:
j-interop-2.06.jar

From what I see, they are using a newer version of the library there:
https://stackoverflow.com/questions/45605656/jinterop-access-is-denied-0x00000005

It might be worth updating it to version 3.0.0:
https://mvnrepository.com/artifact/org.jinterop/j-interop/3.0.0
If you are interested in collaborating in this area, you will be able to test it, we are open.

Best regards.

**Reply by duduyoyo** [reply]
3.0.0 was released 10 years ago and not ready for current patched Windows, IMO.
